lib/logstash/filters/geoip.rb

```diff
@@ -235,4 +237,20 @@ def tag_unsuccessful_lookup(event)
     @tag_on_failure.each{|tag| event.tag(tag)}
   end
 
+  def apply_geodata(geo_data_hash, event)
+    # don't do anything more if the lookup result is nil?
+    return false if geo_data_hash.nil?
+    # only do event.set(@target) if the lookup result is not nil
+    event.set(@target, {}) if event.get(@target).nil?
+    # don't do anything more if the lookup result is empty?
+    return false if geo_data_hash.empty?
+    geo_data_hash.each do |key, value|
+      if @fields.include?(key) && !(value.nil?)
```

Comment:
since we're only excluding Numeric from dup, I assume we never expect value to be either `true` or `false`

that means we can actually replace `!(value.nil)` with just `value`